internal/sci/aws/server.go

```diff
@@ -0,0 +1,218 @@
+	// Fetch the current trust policy
+	getRoleInput := &iam.GetRoleInput{
+		RoleName: awsSdk.String(req.Principal),
+	}
+	getRoleOutput, err := s.Clients.IamClient.GetRole(getRoleInput)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the role: %v", err)
+	}
+
+	// URL decode the trust policy before decoding
+	decodedPolicy, err := url.QueryUnescape(*getRoleOutput.Role.AssumeRolePolicyDocument)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode trust policy: %v", err)
+	}
+
+	// Decode the current trust policy
+	var existingTrustPolicy map[string]interface{}
+	if err := json.Unmarshal([]byte(decodedPolicy), &existingTrustPolicy); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal trust policy: %v", err)
+	}
+
+	subValue := fmt.Sprintf("system:serviceaccount:%s:%s", req.KubernetesNamespace, req.KubernetesServiceAccount)
+
+	// Check if the OIDC provider's trust relationship already exists
+	statements := existingTrustPolicy["Statement"].([]interface{})
+	alreadyExists := false
+	for _, stmt := range statements {
+		stmtMap := stmt.(map[string]interface{})
+		if principal, ok := stmtMap["Principal"].(map[string]interface{}); ok {
+			if federated, ok := principal["Federated"].(string); ok && federated == s.OIDCProviderARN {
+				condition := stmtMap["Condition"].(map[string]interface{})["StringEquals"].(map[string]interface{})
+				condition[fmt.Sprintf("%s:sub", s.OIDCProviderURL)] = subValue
+				alreadyExists = true
+				break
+			}
+		}
+	}
+
+	// Construct the new trust relationship
+	newTrustRelationship := map[string]interface{}{
+		"Effect": "Allow",
+		"Principal": map[string]interface{}{
+			"Federated": s.OIDCProviderARN,
+		},
+		"Action": "sts:AssumeRoleWithWebIdentity",
+		"Condition": map[string]interface{}{
+			"StringEquals": map[string]string{
+				fmt.Sprintf("%s:sub", s.OIDCProviderURL): subValue,
+			},
+		},
+	}
+	if !alreadyExists {
+		// Append the new trust relationship to the existing policy
+		existingTrustPolicy["Statement"] = append(statements, newTrustRelationship)
+	}
+
+	updatedTrustPolicy, err := json.Marshal(existingTrustPolicy)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal updated trust policy: %v", err)
+	}
+
+	// Apply the updated policy
+	input := &iam.UpdateAssumeRolePolicyInput{
+		PolicyDocument: awsSdk.String(string(updatedTrustPolicy)),
+		RoleName:       awsSdk.String(req.Principal),
+	}
+
+	_, err = s.Clients.IamClient.UpdateAssumeRolePolicy(input)
+	if err != nil {
+		return nil, fmt.Errorf("failed to update trust policy: %v", err)
+	}
+
+	return &sci.BindIdentityResponse{}, nil
+}
+
+func GetAccountID(stsSvc *sts.STS) (string, error) {
+	result, err := stsSvc.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	if err == nil {
+		return *result.Account, nil
+	}
+
+	// Fall back to the environment variable if the sts:GetCallerIdentity call fails
+	envAccountID := os.Getenv("AWS_ACCOUNT_ID")
```

Comment:
Huh - I didn't realize that lib existed. It's not in the stdlib but looks like it handles more complex env var use cases. I suppose since we're consuming 2 vars in this SCI, it may be worth it.